src/custom-resource-provider/handler.lambda.ts

```diff
@@ -57,10 +55,24 @@ export const handler: OnEventHandler = async (event: OnEventRequest): Promise<On
 
   const octokit = new Octokit(octokitOptions);
 
+  const parameters = (() => {
+    // Handles the create or update of actions secrets. Retrieves and encrypts the SecretsManager Secret.
+    switch (`${call.endpoint}.${call.method}`) {
+      case "actions.createOrUpdateEnvironmentSecret":
+        return createOrUpdateEnvironmentSecretParameter(octokit, call.parameters);
+      case "actions.createOrUpdateOrgSecret":
+        return createOrUpdateOrgSecretParameter(octokit, call.parameters);
+      case "actions.createOrUpdateRepoSecret":
+        return createOrUpdateRepoSecretParameter(octokit, call.parameters);
+      default:
+        return call.parameters;
+    }
+  })();
+
```

Comment:
Is there a specific reason why the secret stuff is handled in the same lambda?

What do you think if we handle it in a separate lambda? 
I think it could get quite crowded if there happens to pop up more exceptions where you cant just pass all the parameters unchanged to the octokit client.

Reply:
You are totally right.

Handling actions secrets is a different flow than the regular api call:
1. Fetch secret from AWS
1. obtain public key from GitHub
1. encrypt secret
1. put encrypted secret 
1. empty response

If I understand correctly, the best for us would be to handle such use cases in their very own lambdas and they only some common helper logic?

Reply:
Yeah exactly this :)

Reply:
@WtfJoke I move the github api call strategies out into their own file to solve it as fast as possible ... Maybe not the most elegant solution